### PR TITLE
allow getState to be passed to action

### LINKED ISSUE #1: redux-thunk getState support in asyncActionCreator
Nice work here.   I am finding that I am still would like access to redux-thunk's getState argument within my function passed to the asyncActionCreator.  Any advice on this?   I will post a PR with an enhancement allowing this.  In essence we would pass the getState function as the 2nd argument to the action after payload.

Thoughts?

### PR DESCRIPTION
**Pull request by dptoot**
Fixes #1 
Allows access to redux-thunk's getState method within the asyncActionCreator function.

### CODE REVIEW

**Comment by dptoot** [comment]
I have updated my PR to not make any assumptions about what async middleware (redux-thunk, etc...) someone may be using to provide the async action.  As long as dispatch is provided as the first argument by the inner method (which we were assuming before) it now aggregate all arguments of the inner method and passes them to the supplied action.  If you are using redux-thunk, then the argument list for the supplied action is ```js payload, dispatch, getState```

File: src/index.js [hunk 1]
```diff
@@ -17,10 +17,12 @@ export function asyncActionCreator(type, config) {
   if (typeof config === 'function') config = {action: config}; // eslint-disable-line no-param-reassign
   const {client, server, schema} = config;
 
-  return payload => (dispatch, getState) => {
+  return payload => (...args) => {
```
Comment (andy-shea) [inline src/index.js]: Would prefer to use some ES6 niceties here to make it more succinct:
```return payload => (dispatch, ...args) => {```

File: src/index.js [hunk 2]
```diff
@@ -17,10 +17,12 @@ export function asyncActionCreator(type, config) {
...
+    const dispatch = args[0];
```
Comment (andy-shea) [inline src/index.js]: This line can be removed as `dispatch` is made explicit above

File: src/index.js [hunk 3]
```diff
@@ -17,10 +17,12 @@ export function asyncActionCreator(type, config) {
...
     const action = config.action || (isNode ? server : client);
+    args.unshift(payload);
```
Comment (andy-shea) [inline src/index.js]: This line can be removed as `payload` is made explicit below

File: src/index.js [hunk 4]
```diff
@@ -17,10 +17,12 @@ export function asyncActionCreator(type, config) {
...
     dispatch({type, payload});
-    return action(payload, getState).then(
+    return action(...args).then(
```
Comment (andy-shea) [inline src/index.js]: Similarly, this can be changed to 
```return action(payload, dispatch, ...args).then(```
to improve clarity

**Comment by dptoot** [comment]
Good points!  I think I have addressed them.  Would you like me to update any docs as well?

File: src/index.js [hunk 5]
```diff
@@ -17,12 +17,11 @@ export function asyncActionCreator(type, config) {
   if (typeof config === 'function') config = {action: config}; // eslint-disable-line no-param-reassign
   const {client, server, schema} = config;
 
-  return payload => (...args) => {
-    const dispatch = args[0];
+  return payload => (dispatch, ...args) => {
     const action = config.action || (isNode ? server : client);
     args.unshift(payload);
```
Comment (andy-shea) [inline src/index.js]: `args.unshift(payload);` still needs to go; at the moment it will double up the `payload` parameter in the arguments list

**Comment by dptoot** [comment]
How'd I leave that in?  Bad developer! Please let me know if there is anything else you want me to update.

**Comment by andy-shea** [comment]
Looks great, thanks again! :thumbsup:

**Pull request merged**